File: Easy/1678_Goal_Parser_Interpretation/Solution.java



public class Solution {
    public String interpret(String command) {
        StringBuilder result = new StringBuilder();

        for (int i = 0; i < command.length(); ++i) {
            if (command.charAt(i) == 'G') {
                result.append('G');
            } else if (command.charAt(i) == '(' && i + 1 < command.length() && command.charAt(i + 1) == ')') {
                result.append("o");
                i++;
            } else if (command.charAt(i) == '(' && i + 3 < command.length() && command.charAt(i + 1) == 'a' && command.charAt(i + 2) == 'l' && command.charAt(i + 3) == ')') {
                result.append("al");
                i += 3;
            }
        }

        return result.toString();
    }

    public static void main(String[] args) {
        Solution solution = new Solution();

        // test cases 1
        System.out.println(solution.interpret("G()(al)")); // Output: Goal
        
        // test cases 2
        System.out.println(solution.interpret("G()()()()(al)")); // Output: Gooooal

        // test cases 3
        System.out.println(solution.interpret("(al)G(al)()()G")); // Output: alGalooG
    }
}

File: Easy/1678_Goal_Parser_Interpretation/Solution.py
from typing import List, Optional


class Solution:
    def interpret(self, command: str) -> str:
        result = ""

        i = 0
        while i < len(command):
            if command[i] == 'G':
                result += 'G'
            elif command[i] == '(' and i + 1 < len(command) and command[i + 1] == ')':
                result += "o"
                i += 1
            elif command[i] == '(' and i + 3 < len(command) and command[i + 1] == 'a' and command[i + 2] == 'l' and command[i + 3] == ')':
                result += "al"
                i += 3
            i += 1

        return result


if __name__ == "__main__":
    solution = Solution()

    # test cases 1
    print(solution.interpret("G()(al)"))  # Output: Goal
    
    # test cases 2
    print(solution.interpret("G()()()()(al)"))  # Output: Gooooal

    # test cases 3
    print(solution.interpret("(al)G(al)()()G"))  # Output: alGalooG